axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorder/SaleOrderLineProductServiceImpl.java

```diff
@@ -226,8 +224,27 @@ public SaleOrderLine resetProductInformation(SaleOrderLine line) {
     if (appSaleService.getAppSale().getIsEnabledProductDescriptionCopy()) {
       line.setDescription(null);
     }
+    line.setTypeSelect(SaleOrderLineRepository.TYPE_NORMAL);
     line.clearSelectedComplementaryProductList();
-    return line;
+
+    saleOrderLineMap.put("productName", line.getProductName());
+    saleOrderLineMap.put("price", line.getPrice());
+    saleOrderLineMap.put("unit", line.getUnit());
+    saleOrderLineMap.put("companyCostPrice", line.getCompanyCostPrice());
+    saleOrderLineMap.put("discountAmount", line.getDiscountAmount());
+    saleOrderLineMap.put("discountTypeSelect", line.getDiscountTypeSelect());
+    saleOrderLineMap.put("inTaxPrice", line.getInTaxPrice());
+    saleOrderLineMap.put("exTaxTotal", line.getExTaxTotal());
+    saleOrderLineMap.put("inTaxTotal", line.getInTaxTotal());
+    saleOrderLineMap.put("companyInTaxTotal", line.getCompanyInTaxTotal());
+    saleOrderLineMap.put("companyExTaxTotal", line.getCompanyExTaxTotal());
+    saleOrderLineMap.put("description", line.getDescription());
+    saleOrderLineMap.put("typeSelect", line.getTypeSelect());
+    saleOrderLineMap.put(
+        "selectedComplementaryProductList", line.getSelectedComplementaryProductList());
+    saleOrderLineMap.put("taxLineSet", line.getTaxLineSet());
+    saleOrderLineMap.put("taxEquiv", line.getTaxEquiv());
```

Comment:
maybe we could write everything like

```java
line.setFieldName1(value1);
saleOrderLineMap.put("fieldName1", line.getFieldName1());

line.setFieldName2(value2);
saleOrderLineMap.put("fieldName2", line.getFieldName2());
```

so when we will change one field, we will not forget the map ?

Or maybe better, only filling creating and filling the map then using `Mapper.set` method from AOP to update the sale order line.